### PR TITLE
feat: add m-btc on kroma

### PR DESCRIPTION
**Pull request by Martin-ryunseok**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
Free Protocol

##### Twitter Link:
https://twitter.com/FreeLayer2

##### List of audit links if any:
https://docs.meson.fi/references/audit-reports

##### Website Link:
https://app.free.tech/

##### Logo (High resolution, will be shown with rounded borders):
free-protocol

##### Current TVL:
$117,794.8

##### Chain:
Kroma

##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)
merlin-s-seal-btc

##### Short Description (to be shown on DefiLlama):
Free your assets early with http://free.tech. Instantly liquify your staked assets and unlock extra yield potential

##### Token address and ticker if any:
0x0F921c39eFd98809FE6D20a88A4357454578987a

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Bridge

##### methodology (what is being counted as tvl, how is tvl being calculated):
Total Supply of M-BTC

##### Github org/user (Optional, if your code is open source, we can track activity):
https://github.com/CodeToFree

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/free-protocol: 
        
 
  
    
    
    TypeError: addresses.forEach is not a function
    Truncated error stack:
    
        at getUniqueAddresses (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/tokenMapping.js:56:13)
        at sumTokens2 (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/unwrapLPs.js:864:12)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/unwrapLPs.js:984:25
        at getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:29)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:143:15
        at Array.map (<anonymous>)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:131:29
        at Array.map (<anonymous>)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:126:44
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:224:3)
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
        at Object.openSync (node:fs:596:3)
        at Object.writeFileSync (node:fs:2322:35)
        at process.saveCache (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/cache.js:152:17)
        at process.emit (node:events:529:35)
        at process.exit (node:internal/process/per_thread:191:15)
        at process.handleError (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/utils/handleError.js:18:11)
        at process.emit (node:events:529:35)
        at emit (node:internal/process/promises:149:20)
        at processPromiseRejections (node:internal/process/promises:283:27)
        at process.processTicksAndRejections (node:internal/process/task_queues:96:32) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
    }

**Comment by llamatester** [comment]
The adapter at projects/free-protocol exports TVL: 
        
 
 
    merlin                    578.34 M
    bsc                       222.57 M
    ethereum                  10.27 M
    arbitrum                  245.20 k
    manta                     89.77 k
    kroma                     0
    polygon                   0
    
    total                    811.52 M

**Comment by llamatester** [comment]
The adapter at projects/free-protocol exports TVL: 
        
 
 
    merlin                    578.34 M
    bsc                       222.57 M
    ethereum                  10.27 M
    arbitrum                  245.20 k
    manta                     89.77 k
    kroma                     0
    polygon                   0
    
    total                    811.52 M

File: projects/helper/coreAssets.json [hunk 1]
```diff
@@ -1535,7 +1535,8 @@
     "WETH": "0x4200000000000000000000000000000000000001",
     "USDC": "0x49A5010110a358d9069282873F3e7eCf6B41DC10",
     "USDT": "0x0Cf7c2A584988871b654Bd79f96899e4cd6C41C0",
-    "WEMIX": "0x3720b1dc2c8dde3bd6cfcf0b593d0a2bbcac856e"
+    "WEMIX": "0x3720b1dc2c8dde3bd6cfcf0b593d0a2bbcac856e",
+    "M-BTC": "0x0F921c39eFd98809FE6D20a88A4357454578987a"
```
Comment (g1nt0ki) [inline projects/helper/coreAssets.json:1539]: please revert this change

File: projects/free-protocol/index.js [hunk 2]
```diff
@@ -83,4 +83,9 @@ module.exports = {
       ]
     })
   },
+  kroma: {
+    tvl: sumTokensExport({
+      tokens: [ADDRESSES.kroma['M-BTC']]
```
Comment (g1nt0ki) [inline projects/free-protocol/index.js:88]: if nothing locked in kroma but MBTC is minted here, I would recommend changing this line to `tvl: () => ({})`

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
.

File: projects/free-protocol/index.js (hunk 2 above)
Comment (Martin-ryunseok) [reply projects/free-protocol/index.js]: I pushed new commit(#10947)[5712b92](https://github.com/DefiLlama/DefiLlama-Adapters/pull/10947/commits/5712b92639786892c2052402ae243a8548642a46) that applied your comment. Currently, 52 M-BTC already exist on Kroma Chain, having been minted. Additional M-BTC will be minted on Kroma Chain when Free Protocol bridges M-BTC from Merlin Chain to Kroma Chain.

File: projects/helper/coreAssets.json (hunk 1 above)
Comment (Martin-ryunseok) [reply projects/helper/coreAssets.json]: I pushed revert commit [#ab5002a](https://github.com/DefiLlama/DefiLlama-Adapters/pull/10947/commits/ab5002a419a0ba9ce92446d91f2c35cd408cf895)

**Comment by llamatester** [comment]
The adapter at projects/free-protocol exports TVL: 
        
 
 
    merlin                    573.04 M
    bsc                       221.21 M
    ethereum                  10.33 M
    arbitrum                  245.13 k
    manta                     90.37 k
    kroma                     0
    polygon                   0
    
    total                    804.91 M

**Comment by seohoyoon93** [comment]
> The adapter at projects/free-protocol exports TVL:
> 
> ```
> merlin                    573.04 M
> bsc                       221.21 M
> ethereum                  10.33 M
> arbitrum                  245.13 k
> manta                     90.37 k
> kroma                     0
> polygon                   0
> 
> total                    804.91 M 
> ```

@g1nt0ki 

52 M-BTC is already minted on Kroma(https://kromascan.com/token/0x0F921c39eFd98809FE6D20a88A4357454578987a), but the llamatester's exported TVL and the actual data on your service(https://defillama.com/protocol/free-protocol#tvl-charts) does not show any TVL of Kroma.

Could you let us know what should be changed so that DefiLlama can track the TVL of M-BTC on Kroma?
